lib/src/Pos/Logic/Types.hs

```diff
@@ -27,53 +25,39 @@ import           Pos.Util.Chrono (NE, NewestFirst, OldestFirst)
 -- | The interface to a logic layer, i.e. some component which encapsulates
 -- blockchain / crypto logic.
 data Logic m = Logic
-    { -- The stakeholder id of our node.
+    { -- | The stakeholder id of our node.
       ourStakeholderId   :: StakeholderId
-      -- Get a block, perhaps from a database.
+      -- | Get a block, perhaps from a database.
     , getBlock           :: HeaderHash -> m (Maybe Block)
-      -- Stream blocks from first hash to second hash.
-      -- Conduit is chosen mainly due to precedent: it's already used in
-      -- cardano-sl.
-    , getChainFrom       :: HeaderHash
-                         -> Source m Block
-      -- Get a block header.
-      -- TBD: necessary? Is it any different/faster than getting the block
-      -- and taking the header?
+      -- | Get a block header.
     , getBlockHeader     :: HeaderHash -> m (Maybe BlockHeader)
-      -- Inspired by 'getHeadersFromManyTo'.
-      -- Included here because that function is quite complicated; it's not
-      -- clear whether it can be expressed simply in terms of getBlockHeader.:q
+      -- TODO CSL-2089 use conduits in this and the following methods
+      -- | Retrieve block header hashes from specified interval.
+    , getHashesRange     :: Maybe Word -- ^ Optional limit on how many to bring in.
+                         -> HeaderHash
+                         -> HeaderHash
+                         -> m (OldestFirst NE HeaderHash)
```

Comment:
Change it to `Either GetHashesRangeError (OldestFirst NE HeaderHash)` or `Maybe (OldestFirst NE HeaderHash)`.